src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -476,6 +476,13 @@ private OidcConfiguration buildOidcConfiguration() {
         // TODO what do we prefer?
         // conf.setPreferredJwsAlgorithm(JWSAlgorithm.HS256);
```

Comment:
Wherever we set preferredJwsAlgo, we need to call the validation method.

@jtnord  - I assume this will be uncommented and algorithm will be passed dynamically